crates/core/src/syn/parser/kind.rs

```diff
@@ -145,6 +146,28 @@ impl Parser<'_> {
 					Ok(Kind::Set(Box::new(Kind::Any), None))
 				}
 			}
+			t!("REFERENCES") => {
+				if !self.settings.references_enabled {
+					bail!("Experimental capability `record_references` is not enabled")
```

Comment:
Missing span information for the error here, best to add a span so that the user is being pointed to where the reference keyword is used. 
```suggestion
					bail!("Experimental capability `record_references` is not enabled", @self.last_span() => "Use of `REFERENCES` keyword is still experimental")
```